lib/reverse_sentence.rb

```diff
@@ -1,6 +1,57 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^2) where n is the length of my_sentence
+# Space complexity: Constant or O(1) because we are not creating any new data
+#                   structures and therefore not taking up more space in memory
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  my_sentence = reverse_words(my_sentence)
+  return string_reverse(my_sentence)
+  
+end
+
+def reverse_words(my_words)
+  return nil if my_words.nil?
+
+  i = 0
```

Comment:
For consistency with the method `reverse(array, low, high)` I would recommend updating these variable names to be more descriptive.